// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/ApplicationSecurityWafCustomRuleConditionInputAddress.js
"use strict";
/**
 * Unless explicitly stated otherwise all files in this repository are licensed under the Apache-2.0 License.
 * This product includes software developed at Datadog (https://www.datadoghq.com/).
 * Copyright 2020-Present Datadog, Inc.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.HTTP_CLIENT_IP = exports.USR_ID = exports.GRAPHQL_SERVER_ALL_RESOLVERS = exports.GRPC_SERVER_METHOD = exports.GRPC_SERVER_REQUEST_MESSAGE = exports.GRPC_SERVER_REQUEST_METADATA = exports.SERVER_RESPONSE_TRAILERS = exports.SERVER_RESPONSE_HEADERS_NO_COOKIES = exports.SERVER_RESPONSE_STATUS = exports.SERVER_REQUEST_BODY = exports.SERVER_REQUEST_TRAILERS = exports.SERVER_REQUEST_COOKIES = exports.SERVER_REQUEST_HEADERS_NO_COOKIES = exports.SERVER_REQUEST_QUERY = exports.SERVER_REQUEST_PATH_PARAMS = exports.SERVER_REQUEST_URI_RAW = exports.SERVER_REQUEST_METHOD = exports.SERVER_SYS_SHELL_CMD = exports.SERVER_IO_NET_URL = exports.SERVER_IO_FS_FILE = exports.SERVER_DB_STATEMENT = void 0;
exports.SERVER_DB_STATEMENT = "server.db.statement";
exports.SERVER_IO_FS_FILE = "server.io.fs.file";
exports.SERVER_IO_NET_URL = "server.io.net.url";
exports.SERVER_SYS_SHELL_CMD = "server.sys.shell.cmd";
exports.SERVER_REQUEST_METHOD = "server.request.method";
exports.SERVER_REQUEST_URI_RAW = "server.request.uri.raw";
exports.SERVER_REQUEST_PATH_PARAMS = "server.request.path_params";
exports.SERVER_REQUEST_QUERY = "server.request.query";
exports.SERVER_REQUEST_HEADERS_NO_COOKIES = "server.request.headers.no_cookies";
exports.SERVER_REQUEST_COOKIES = "server.request.cookies";
exports.SERVER_REQUEST_TRAILERS = "server.request.trailers";
exports.SERVER_REQUEST_BODY = "server.request.body";
exports.SERVER_RESPONSE_STATUS = "server.response.status";
exports.SERVER_RESPONSE_HEADERS_NO_COOKIES = "server.response.headers.no_cookies";
exports.SERVER_RESPONSE_TRAILERS = "server.response.trailers";
exports.GRPC_SERVER_REQUEST_METADATA = "grpc.server.request.metadata";
exports.GRPC_SERVER_REQUEST_MESSAGE = "grpc.server.request.message";
exports.GRPC_SERVER_METHOD = "grpc.server.method";
exports.GRAPHQL_SERVER_ALL_RESOLVERS = "graphql.server.all_resolvers";
exports.USR_ID = "usr.id";
exports.HTTP_CLIENT_IP = "http.client_ip";
