// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/ScaRequestDataAttributesDependenciesItemsLocationsItems.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.ScaRequestDataAttributesDependenciesItemsLocationsItems = void 0;
class ScaRequestDataAttributesDependenciesItemsLocationsItems {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return ScaRequestDataAttributesDependenciesItemsLocationsItems.attributeTypeMap;
    }
}
exports.ScaRequestDataAttributesDependenciesItemsLocationsItems = ScaRequestDataAttributesDependenciesItemsLocationsItems;
/**
 * @ignore
 */
ScaRequestDataAttributesDependenciesItemsLocationsItems.attributeTypeMap = {
    block: {
        baseName: "block",
        type: "ScaRequestDataAttributesDependenciesItemsLocationsItemsFilePosition",
    },
    name: {
        baseName: "name",
        type: "ScaRequestDataAttributesDependenciesItemsLocationsItemsFilePosition",
    },
    namespace: {
        baseName: "namespace",
        type: "ScaRequestDataAttributesDependenciesItemsLocationsItemsFilePosition",
    },
    version: {
        baseName: "version",
        type: "ScaRequestDataAttributesDependenciesItemsLocationsItemsFilePosition",
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
