// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/ScaRequestDataAttributesDependenciesItemsLocationsItemsFilePosition.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.ScaRequestDataAttributesDependenciesItemsLocationsItemsFilePosition = void 0;
class ScaRequestDataAttributesDependenciesItemsLocationsItemsFilePosition {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return ScaRequestDataAttributesDependenciesItemsLocationsItemsFilePosition.attributeTypeMap;
    }
}
exports.ScaRequestDataAttributesDependenciesItemsLocationsItemsFilePosition = ScaRequestDataAttributesDependenciesItemsLocationsItemsFilePosition;
/**
 * @ignore
 */
ScaRequestDataAttributesDependenciesItemsLocationsItemsFilePosition.attributeTypeMap = {
    end: {
        baseName: "end",
        type: "ScaRequestDataAttributesDependenciesItemsLocationsItemsPosition",
    },
    fileName: {
        baseName: "file_name",
        type: "string",
    },
    start: {
        baseName: "start",
        type: "ScaRequestDataAttributesDependenciesItemsLocationsItemsPosition",
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
